src/crypto-providers/ledgerCryptoProvider.ts

```diff
@@ -253,6 +292,23 @@ export const LedgerCryptoProvider: () => Promise<CryptoProvider> = async () => {
     }
   }
 
+  const prepareTtl = (ttl?: number) => {
```

Comment:
ttl should technically be a BigInt, not number, Ledger/CDDL specifies it as int64

Reply:
also please specify return type, I guess it's `string?` (ideally `string | null` IMHO to avoid dealing with undefined but I guess optionals are all around so this would be a bit futile)

Reply:
Fixed